app/src/visionglass/java/com/igalia/wolvic/PlatformActivity.java

```diff
@@ -126,6 +130,33 @@ protected Intent getStoreIntent() {
     };
     private final Runnable activityResumedRunnable = this::activityResumed;
 
+    // Unfortunately, the usual solutions for keeping the screen active (WakeLock, KEEP_SCREEN_ON)
+    // do not work when the Vision Glass is connected, so we need to simulate a touch periodically.
+    private final Handler mMainHandler = new Handler(Looper.getMainLooper());
+    private final String KEEP_SCREEN_ON_COMMAND = "input touchscreen tap 1 1";
+    private final int KEEP_SCREEN_ON_DELAY = 8_000;
```

Comment:
This is kind of random. You can get the actual screen lock time in ms using
Settings.System.getInt(getContentResolver(), Settings.System.SCREEN_OFF_TIMEOUT, 30000);

Note that in my experience the screen is locked earlier, in my testing you have to adjust that value by 2/3 approx.

Reply:
The reason is that 8 seconds is a bit over half of the minimum timeout that can be set on the phone (15 seconds).

One issue with fetching that value dynamically is that it might change it while the application is running. For example, the phone might change it automatically when entering power saving mode.

I will increase a bit the delay to 10 seconds, which IMHO seems a reasonable compromise.

Reply:
In order to change it the user must switch to the settings app meaning that Wolvic will be paused. So you can always refetch that value on resuming. That should be safe enough.